src/category_theory/abelian/diagram_lemmas/four.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import category_theory.abelian.pseudoelements
+
+/-!
+# The four lemma
+
+We prove the "mono" version of the four lemma in an arbitrary abelian category.
+
+## Future work
+
+The "epi" four lemma and the five lemma, which is then an easy corollary.
+
+## Tags
+
+four lemma, diagram lemma, diagram chase
+-/
+open category_theory
+open category_theory.abelian.pseudoelements
+
+universes v u
+
+variables {V : Type u} [category.{v} V] [abelian V]
+
+local attribute [instance] preadditive.has_equalizers_of_has_kernels
+local attribute [instance] object_to_sort hom_to_fun
+
+section four
+
+/-
+
+A ---f--> B ---g--> C ---h--> D
+|         |         |         |
+α         β         γ         δ
+|         |         |         |
+v         v         v         v
+A' --f'-> B' --g'-> C' --h'-> D'
+
+-/
+
+variables {A B C D A' B' C' D' : V}
+variables {f : A ⟶ B} {g : B ⟶ C} {h : C ⟶ D}
+variables {f' : A' ⟶ B'} {g' : B' ⟶ C'} {h' : C' ⟶ D'}
+variables {α : A ⟶ A'} {β : B ⟶ B'} {γ : C ⟶ C'} {δ : D ⟶ D'}
+variables [exact f g] [exact g h] [exact f' g']
+variables (comm₁ : α ≫ f' = f ≫ β) (comm₂ : β ≫ g' = g ≫ γ) (comm₃ : γ ≫ h' = h ≫ δ)
+include comm₁ comm₂ comm₃
+
+/-- The four lemma, mono version -/
```

Comment:
I think it would be very helpful for users if the ascii-art diagram appears in the docstring.